File: Arbitrary_Precision/Java/CCExtras.java
/*
 * SPDX-License-Identifier: Apache-2.0 OR LGPL-3.0-or-later
 *
 * This Source Code Form is subject to the terms of the Apache License,
 * v. 2.0.
 * If a copy of the Apache License was not distributed with this file,
 * You can obtain one at http://www.apache.org/licenses/LICENSE-2.0.
 *
 * Alternatively, this Source Code Form is subject to the terms of the
 * GNU Lesser General Public License, v. 3.0 or later. If a copy of the
 * LGPL was not distributed with this file, You can obtain one at
 * https://www.gnu.org/licenses/lgpl-3.0.html.
 *
 * Copyright 2026 Skylar Koningin
 */

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.charset.StandardCharsets;
import java.nio.file.StandardOpenOption;

public class CCExtras {
    
    public static void writeToFile(Path filePath, String text, boolean append) {
        if (append) {
            try {
                Files.write(filePath, text.getBytes(StandardCharsets.UTF_8), 
                     StandardOpenOption.APPEND, StandardOpenOption.CREATE);
            } catch (IOException e) {
                e.printStackTrace();
            }
        } else {
            try {
                Files.writeString(filePath, text);
            } catch (IOException e) {
                e.printStackTrace();
            }
        }
    }
    
    
}

File: Arbitrary_Precision/Ruby/cc_extras.rb
# SPDX-License-Identifier: Apache-2.0 OR LGPL-3.0-or-later
#
# This Source Code Form is subject to the terms of the Apache License,
# v. 2.0.
# If a copy of the Apache License was not distributed with this file,
# You can obtain one at http://www.apache.org/licenses/LICENSE-2.0.
#
# Alternatively, this Source Code Form is subject to the terms of the
# GNU Lesser General Public License, v. 3.0 or later. If a copy of the
# LGPL was not distributed with this file, You can obtain one at
# https://www.gnu.org/licenses/lgpl-3.0.html.
#
# Copyright 2026 Skylar Koningin


# frozen_string_literal: true

# fuck ts fr
module CCExtras
  def self.write_to_file(file_path, text, append)
    if append
      File.write file_path, text, mode: 'a'
    else
      File.write file_path, text
    end
  end
end
